GTG/gtk/editor/recurring_menu.py

```diff
@@ -0,0 +1,88 @@
+from datetime import datetime
+class RecurringMenu():
+    """ RecurringMenu provides a simple layer of abstraction
+    for the menu where the user enables a task to be repeating
+    """
+    PREFIX = "Every "
+
+    def __init__(self, requester, tid, builder):
+        # General attributes
+        self.task = requester.get_task(tid)
+        self.selected_recurring_term = self.task.get_recurring_term()
+        
+        # Getting the necessary Gtk objects
+        self.title = builder.get_object("title_label")
+        self.title_separator = builder.get_object("title_separator")
+        self.repeat_button = builder.get_object("repeat_checkbutton")
+        self.repeat_icon = builder.get_object("repeat_icon")
+        self.icon_style = self.repeat_icon.get_style_context()
+        self.stack = builder.get_object("main_stack")
+        self.page1 = builder.get_object("stack_main_box")
+
+        # Update the editor using the task recurring status
+        self.update_header()
+        self.repeat_button.set_active(self.task.get_recurring())
+        if self.task.get_recurring():
+            self.icon_style.add_class('recurring-active')
+
+    def update_repeat_button(self, active=True):
+        if active:
+            self.icon_style.add_class('recurring-active')
+        else:
+            self.icon_style.remove_class('recurring-active')
+
+    def is_term_set(self):
+        return self.selected_recurring_term is not None
+
+    def set_selected_term(self, string):
+        self.selected_recurring_term = string
+
+    def update_tick(self):
```

Comment:
Does tick refer to the checkbox? I'd rather use the same terms Gtk uses, like `update_checkbox()`